File: src/main/java/io/github/kingg22/api_vacunas_panama/web/dto/ProvinciaDto.java
package io.github.kingg22.api_vacunas_panama.web.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.kingg22.api_vacunas_panama.persistence.entity.Provincia;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import org.springframework.validation.annotation.Validated;

import java.io.Serializable;

/**
 * DTO for {@link Provincia}
 */
@Validated
public record ProvinciaDto(Short id,
                           @NotNull
                           @Size(max = 30)
                           @JsonProperty(defaultValue = "Por registrar")
                           String nombre) implements Serializable {
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/direccion/dto/ProvinciaDto.kt
package io.github.kingg22.api.vacunas.panama.modules.direccion.dto

import io.github.kingg22.api.vacunas.panama.modules.direccion.domain.ProvinciaModel
import io.github.kingg22.api.vacunas.panama.modules.direccion.entity.Provincia
import io.mcarle.konvert.api.KonvertTo
import io.quarkus.runtime.annotations.RegisterForReflection
import jakarta.validation.constraints.Size
import java.io.Serializable

/** DTO for [io.github.kingg22.api.vacunas.panama.modules.direccion.entity.Provincia] */
@RegisterForReflection
@KonvertTo(Provincia::class)
@KonvertTo(ProvinciaModel::class)
@JvmRecord
data class ProvinciaDto(
    val id: Short? = null,

    @all:Size(max = 30)
    val nombre: String = DEFAULT_PROVINCIA,
) : Serializable {
    companion object {
        const val DEFAULT_PROVINCIA = "Por registrar"
    }
}
